web/renderer/components/CloneDatabaseForm/CloneForm.tsx

```diff
@@ -4,26 +4,30 @@ import {
   FormInput,
   Popup,
 } from "@dolthub/react-components";
-import { connections as connectionsUrl } from "@lib/urls";
-import { DatabaseConnectionFragment } from "@gen/graphql-types";
-import Link from "@components/links/Link";
-import { ConfigState } from "./context/state";
-import { useConfigContext } from "./context/config";
+import { ConfigState } from "@components/pageComponents/ConnectionsPage/NewConnection/context/state";
 import css from "./index.module.css";
+import { getStartLocalDoltServerDisabled } from "./utils";
+import { useClone } from "./useClone";
 
-export default function CloneDoltDatabaseForm() {
+type Props = {
+  connectionState: ConfigState;
```

Comment:
This is much closer, but instead of passing this big state around and handling both pages in `onCloneDoltHubDatabase`, can you pass `onCloneDoltHubDatabase` as a prop and it can have whatever state you need from `CloneForm` (maybe just database and owner name) as arguments? Like:
```suggestion
  onCloneDoltHubDatabase: (e: SyntheticEvent, owner: string, database: string) => Promise<void>;
```

And since most of this state is only needed in the new connections page it can stay there and each page can handle its own clone logic